### PR TITLE
`azurerm_cdn_endpoint` extensions

### PR DESCRIPTION
**Pull request by tombuildsstuff**
- [x] Support for Geo Filters and ProbePaths
- [x] Refactoring
- [x] ~Determine if a state migration is needed for the `origin` block~ doesn't look like it
- [x] Makes the `origin` block properties `ForceNew`

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
Tests pass:

screen shot 2018-03-12 at 17 32 09

File: azurerm/resource_arm_cdn_endpoint.go [hunk 1]
```diff
@@ -137,75 +191,131 @@ func resourceArmCdnEndpointCreate(d *schema.ResourceData, meta interface{}) erro
...
+
+	return resourceArmCdnEndpointRead(d, meta)
+}
+
+func resourceArmCdnEndpointUpdate(d *schema.ResourceData, meta interface{}) error {
+	endpointsClient := meta.(*ArmClient).cdnEndpointsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
```
Comment (katbyte) [inline azurerm/resource_arm_cdn_endpoint.go]: This was changed to resourceGroup above.

File: website/docs/r/cdn_endpoint.html.markdown [hunk 2]
```diff
@@ -51,44 +51,55 @@ resource "azurerm_cdn_endpoint" "test" {
...
+* `origin_path` - (Optional) The path used at for origin requests.
+
+* `probe_path` - (Optional) the path to a file hosted on the origin which helps accelerate delivery of the dynamic content and calculate the most optimal routes for the CDN. This is relative to the `origin_path`.
 
 * `tags` - (Optional) A mapping of tags to assign to the resource.
 
 The `origin` block supports:
 
-* `name` - (Required) The name of the origin. This is an arbitrary value. However, this value needs to be unique under endpoint.
+* `name` - (Required) The name of the origin. This is an arbitrary value. However, this value needs to be unique under endpoint. Changing this forces a new resource to be created.
```
Comment (katbyte) [inline website/docs/r/cdn_endpoint.html.markdown]: I think there is a missing ‘the’ in ‘unique under the endpoint’

Comment (tombuildsstuff) [reply website/docs/r/cdn_endpoint.html.markdown]: Good spot - fixed.

File: azurerm/resource_arm_cdn_endpoint.go (hunk 1 above)
Comment (tombuildsstuff) [reply azurerm/resource_arm_cdn_endpoint.go]: 👍 fixed

**Pull request merged**